src/compiler/checker.ts

```diff
@@ -566,7 +566,7 @@ module ts {
         }
 
         function getTargetOfExportAssignment(node: ExportAssignment): Symbol {
-            return resolveEntityName(<Identifier>node.expression, SymbolFlags.Value | SymbolFlags.Type | SymbolFlags.Namespace);
+            return node.expression && resolveEntityName(<Identifier>node.expression, SymbolFlags.Value | SymbolFlags.Type | SymbolFlags.Namespace);

```

Comment:
I understand what you are saying about the binder. But I don't understand how it implies that over here, you can assume node.expression is an Identifier.